smallrye-reactive-messaging-provider/src/main/java/io/smallrye/reactive/messaging/impl/ConnectorConfig.java

```diff
@@ -131,47 +159,56 @@ public ConfigValue getConfigValue(String propertyName) {
      */
     @Override
     public Iterable<String> getPropertyNames() {
-        String connectorPrefix = CONNECTOR_PREFIX + connector + ".";
         String prefix = this.prefix + name + ".";
-        String prefixFromEnv = toEnv(prefix);
-        String connectorPrefixFromEnv = toEnv(connectorPrefix);
+        String prefixAlpha = toAlpha(prefix);
+        String prefixAlphaUpper = prefixAlpha.toUpperCase();
+        String connectorPrefix = CONNECTOR_PREFIX + connector + ".";
+        String connectorPrefixAlpha = toAlpha(connectorPrefix);
+        String connectorPrefixAlphaUpper = connectorPrefixAlpha.toUpperCase();
 
         Set<String> names = new HashSet<>();
         for (String name : overall.getPropertyNames()) {
             if (name.startsWith(connectorPrefix)) {
                 String computed = name.substring(connectorPrefix.length());
-                if (doesNotContainEnv(names, computed)) {
+                names.add(computed);
+            } else if (name.startsWith(connectorPrefixAlpha)) {
+                String computed = name.substring(connectorPrefixAlpha.length());
+                if (nameExists(connectorPrefix + computed)) {
+                    names.add(computed);
+                }
+            } else if (name.startsWith(connectorPrefixAlphaUpper)) {
+                String computed = name.substring(connectorPrefixAlphaUpper.length());
+                if (nameExists(connectorPrefix + computed)) {
                     names.add(computed);
                 }
-            } else if (name.startsWith(connectorPrefixFromEnv)) {
-                String computed = name.substring(connectorPrefixFromEnv.length());
-                // Remove the potential existing key
-                names.removeIf(s -> toEnv(s).equalsIgnoreCase(computed));
-                names.add(computed);
             } else if (name.startsWith(prefix)) {
                 String computed = name.substring(prefix.length());
-                if (doesNotContainEnv(names, computed)) {
+                names.add(computed);
+            } else if (name.startsWith(prefixAlpha)) {
+                String computed = name.substring(prefixAlpha.length());
+                if (nameExists(prefix + computed)) {
+                    names.add(computed);
+                }
+            } else if (name.startsWith(prefixAlphaUpper)) {
+                String computed = name.substring(prefixAlphaUpper.length());
+                if (nameExists(prefix + computed)) {
                     names.add(computed);
                 }
-            } else if (name.startsWith(prefixFromEnv)) {
-                String computed = name.substring(prefixFromEnv.length());
-                // Remove the potential existing key
-                names.removeIf(s -> toEnv(s).equalsIgnoreCase(computed));
-                names.add(computed);
             }
```

Comment:
I don't really like this code but I couldn't find a better way to do it.

The problem is that config properties derived from environment variables should be returned if they match the confit name exactly, match the config name with non-alphanumeric characters converted to `_` or match the config name with non-alphanumber caracters converted to `_` and the rest converted to upper case.

However, for config properties from other sources, they usually have to match exactly.

The only way I could find to do this was to first try all the combinations which are allowed by the environment variable config source, then check if the final name we've created is actually present in the config. If it was from an environment variable we should find it, but if it wasn't then we won't.

Reply:
I also removed the behaviour where you tried to avoid adding a name twice if it only differred by case, since the smallrye config implementation would return both names.

(e.g. if I have `foo` in a microprofile-config.properties file, and `FOO` in an environement variable, `getPropertyNames()` will return both `foo` and `FOO`, even though I'll get the same value when I look either of them up.)

Reply:
Unfortunately, if we did that it's because we got a bug where the "foo" attribute was unable to be overridden by "FOO".
If I remember correctly it was:

`kafka.bootstrap.servers` vs. `KAFKA_BOOTSTRAP_SERVERS`. 

So we must be sure that this works (as well as when `kafka.bootstrap.servers` is not set and uses the default)